target_postgres/connector.py

```diff
@@ -30,6 +31,37 @@ def create_sqlalchemy_connection(self) -> sqlalchemy.engine.Connection:
         """
         return self.create_sqlalchemy_engine().connect()
 
+    def get_sqlalchemy_url(self, config: dict) -> str:
+        """Generates a SQLAlchemy URL for sqlbuzz.
+        Args:
+            config: The configuration for the connector.
+        """
+        if config['dialect'] == "postgresql" :
+            url_drivername:str = config['dialect']
+        else:
+            self.logger.error("Invalid dialect given")
+            exit(1)
+
+        if config['driver_type'] in ["psycopg2","pg8000","asyncpg","psycopg2cffi","pypostgresql","pygresql"]:
+            url_drivername += f"+{config['driver_type']}"
+        else:
+            self.logger.error("Invalid driver_type given")
+            exit(1)
+
+        self.logger.info(url_drivername)
+        config_url = URL.create(
```

Comment:
I'm a little worried that we aren't flexible enough here. I think allowing a sqlalchemy url override solves my issue. There's probably options for authing over TLS and other auth options, and connection options available and we're being pretty strict about how we want you to connect to the db.